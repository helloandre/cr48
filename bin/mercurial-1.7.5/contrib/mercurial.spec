Summary: A fast, lightweight Source Control Management system
Name: mercurial
Version: snapshot
Release: 0
License: GPLv2+
Group: Development/Tools
URL: http://mercurial.selenic.com/
Source0: http://mercurial.selenic.com/release/%{name}-%{version}.tar.gz
BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root

# From the README:
#
#   Note: some distributions fails to include bits of distutils by
#   default, you'll need python-dev to install. You'll also need a C
#   compiler and a 3-way merge tool like merge, tkdiff, or kdiff3.
#
# python-devel provides an adequate python-dev.  The merge tool is a
# run-time dependency.
#
BuildRequires: python >= 2.4, python-devel, make, gcc, python-docutils >= 0.5, gettext
Provides: hg = %{version}-%{release}
Requires: python >= 2.4
# The hgk extension uses the wish tcl interpreter, but we don't enforce it
#Requires: tk

%define pythonver %(python -c 'import sys;print ".".join(map(str, sys.version_info[:2]))')
%define emacs_lispdir %{_datadir}/emacs/site-lisp

%description
Mercurial is a fast, lightweight source control management system designed
for efficient handling of very large distributed projects.

%prep
%setup -q

%build
make all

%install
rm -rf $RPM_BUILD_ROOT
make install DESTDIR=$RPM_BUILD_ROOT PREFIX=%{_prefix} MANDIR=%{_mandir}

install -m 755 contrib/hgk $RPM_BUILD_ROOT%{_bindir}
install -m 755 contrib/hg-ssh $RPM_BUILD_ROOT%{_bindir}

bash_completion_dir=$RPM_BUILD_ROOT%{_sysconfdir}/bash_completion.d
mkdir -p $bash_completion_dir
install -m 644 contrib/bash_completion $bash_completion_dir/mercurial.sh

zsh_completion_dir=$RPM_BUILD_ROOT%{_datadir}/zsh/site-functions
mkdir -p $zsh_completion_dir
install -m 644 contrib/zsh_completion $zsh_completion_dir/_mercurial

mkdir -p $RPM_BUILD_ROOT%{emacs_lispdir}
install -m 644 contrib/mercurial.el $RPM_BUILD_ROOT%{emacs_lispdir}
install -m 644 contrib/mq.el $RPM_BUILD_ROOT%{emacs_lispdir}

mkdir -p $RPM_BUILD_ROOT/%{_sysconfdir}/mercurial/hgrc.d
install -m 644 contrib/mergetools.hgrc $RPM_BUILD_ROOT%{_sysconfdir}/mercurial/hgrc.d/mergetools.rc

%clean
rm -rf $RPM_BUILD_ROOT

%files
%defattr(-,root,root,-)
%doc CONTRIBUTORS COPYING doc/README doc/hg*.txt doc/hg*.html *.cgi contrib/*.fcgi
%doc %attr(644,root,root) %{_mandir}/man?/hg*
%doc %attr(644,root,root) contrib/*.svg contrib/sample.hgrc
%dir %{_datadir}/zsh/
%dir %{_datadir}/zsh/site-functions/
%{_datadir}/zsh/site-functions/_mercurial
%dir %{_datadir}/emacs/site-lisp/
%{_datadir}/emacs/site-lisp/mercurial.el
%{_datadir}/emacs/site-lisp/mq.el
%{_bindir}/hg
%{_bindir}/hgk
%{_bindir}/hg-ssh
%dir %{_sysconfdir}/bash_completion.d/
%config(noreplace) %{_sysconfdir}/bash_completion.d/mercurial.sh
%dir %{_sysconfdir}/mercurial
%dir %{_sysconfdir}/mercurial/hgrc.d
%config(noreplace) %{_sysconfdir}/mercurial/hgrc.d/mergetools.rc
%if "%{?pythonver}" != "2.4"
%{_libdir}/python%{pythonver}/site-packages/%{name}-*-py%{pythonver}.egg-info
%endif
%{_libdir}/python%{pythonver}/site-packages/%{name}
%{_libdir}/python%{pythonver}/site-packages/hgext
