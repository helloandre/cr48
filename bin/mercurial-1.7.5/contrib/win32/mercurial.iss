; Script generated by the Inno Setup Script Wizard.
; SEE THE DOCUMENTATION FOR DETAILS ON CREATING INNO SETUP SCRIPT FILES!

#ifndef VERSION
#define FileHandle
#define FileLine
#define VERSION = "unknown"
#if FileHandle = FileOpen(SourcePath + "\..\..\mercurial\__version__.py")
  #expr FileLine = FileRead(FileHandle)
  #expr FileLine = FileRead(FileHandle)
  #define VERSION = Copy(FileLine, Pos('"', FileLine)+1, Len(FileLine)-Pos('"', FileLine)-1)
#endif
#if FileHandle
  #expr FileClose(FileHandle)
#endif
#pragma message "Detected Version: " + VERSION
#endif

#ifndef ARCH
#define ARCH = "x86"
#endif

[Setup]
AppCopyright=Copyright 2005-2010 Matt Mackall and others
AppName=Mercurial
#if ARCH == "x64"
AppVerName=Mercurial {#VERSION} (64-bit)
OutputBaseFilename=Mercurial-{#VERSION}-x64
ArchitecturesAllowed=x64
ArchitecturesInstallIn64BitMode=x64
#else
AppVerName=Mercurial {#VERSION}
OutputBaseFilename=Mercurial-{#VERSION}
#endif
InfoAfterFile=contrib/win32/postinstall.txt
LicenseFile=COPYING
ShowLanguageDialog=yes
AppPublisher=Matt Mackall and others
AppPublisherURL=http://mercurial.selenic.com/
AppSupportURL=http://mercurial.selenic.com/
AppUpdatesURL=http://mercurial.selenic.com/
AppID={{4B95A5F1-EF59-4B08-BED8-C891C46121B3}
AppContact=mercurial@selenic.com
DefaultDirName={pf}\Mercurial
SourceDir=..\..
VersionInfoDescription=Mercurial distributed SCM (version {#VERSION})
VersionInfoCopyright=Copyright 2005-2010 Matt Mackall and others
VersionInfoCompany=Matt Mackall and others
InternalCompressLevel=max
SolidCompression=true
SetupIconFile=contrib\win32\mercurial.ico
AllowNoIcons=true
DefaultGroupName=Mercurial
PrivilegesRequired=none

[Files]
Source: contrib\mercurial.el; DestDir: {app}/Contrib
Source: contrib\vim\*.*; DestDir: {app}/Contrib/Vim
Source: contrib\zsh_completion; DestDir: {app}/Contrib
Source: contrib\bash_completion; DestDir: {app}/Contrib
Source: contrib\tcsh_completion; DestDir: {app}/Contrib
Source: contrib\tcsh_completion_build.sh; DestDir: {app}/Contrib
Source: contrib\hgk; DestDir: {app}/Contrib; DestName: hgk.tcl
Source: contrib\xml.rnc; DestDir: {app}/Contrib
Source: contrib\shrink-revlog.py; DestDir: {app}/Contrib
Source: contrib\mercurial.el; DestDir: {app}/Contrib
Source: contrib\mq.el; DestDir: {app}/Contrib
Source: contrib\hgweb.fcgi; DestDir: {app}/Contrib
Source: contrib\hgweb.wsgi; DestDir: {app}/Contrib
Source: contrib\win32\ReadMe.html; DestDir: {app}; Flags: isreadme
Source: contrib\mergetools.hgrc; DestDir: {tmp};
Source: contrib\win32\mercurial.ini; DestDir: {app}; DestName: Mercurial.ini; Check: CheckFile; AfterInstall: ConcatenateFiles;
Source: contrib\win32\postinstall.txt; DestDir: {app}; DestName: ReleaseNotes.txt
Source: dist\hg.exe; DestDir: {app}; AfterInstall: Touch('{app}\hg.exe.local')
#if ARCH == "x64"
Source: dist\*.dll; Destdir: {app}
Source: dist\*.pyd; Destdir: {app}
#else
Source: dist\python*.dll; Destdir: {app}; Flags: skipifsourcedoesntexist
Source: dist\msvc*.dll; DestDir: {app}; Flags: skipifsourcedoesntexist
Source: dist\w9xpopen.exe; DestDir: {app}
#endif
Source: dist\Microsoft.VC*.CRT.manifest; DestDir: {app}; Flags: skipifsourcedoesntexist
Source: dist\library.zip; DestDir: {app}
Source: dist\add_path.exe; DestDir: {app}
Source: dist\cacert.pem; Destdir: {app}
Source: doc\*.html; DestDir: {app}\Docs
Source: doc\style.css; DestDir: {app}\Docs
Source: mercurial\help\*.txt; DestDir: {app}\help
Source: mercurial\locale\*.*; DestDir: {app}\locale; Flags: recursesubdirs createallsubdirs skipifsourcedoesntexist
Source: mercurial\templates\*.*; DestDir: {app}\Templates; Flags: recursesubdirs createallsubdirs
Source: CONTRIBUTORS; DestDir: {app}; DestName: Contributors.txt
Source: COPYING; DestDir: {app}; DestName: Copying.txt

[INI]
Filename: {app}\Mercurial.url; Section: InternetShortcut; Key: URL; String: http://mercurial.selenic.com/
Filename: {app}\Mercurial.ini; Section: web; Key: cacerts; String: {app}\cacert.pem

[UninstallDelete]
Type: files; Name: {app}\Mercurial.url
Type: files; Name: {app}\Contrib\shrink-revlog.pyc

[Icons]
Name: {group}\Uninstall Mercurial; Filename: {uninstallexe}
Name: {group}\Mercurial Command Reference; Filename: {app}\Docs\hg.1.html
Name: {group}\Mercurial Configuration Files; Filename: {app}\Docs\hgrc.5.html
Name: {group}\Mercurial Ignore Files; Filename: {app}\Docs\hgignore.5.html
Name: {group}\Mercurial Web Site; Filename: {app}\Mercurial.url

[Run]
Filename: "{app}\add_path.exe"; Parameters: "{app}"; Flags: postinstall; Description: "Add the installation path to the search path"

[UninstallRun]
Filename: "{app}\add_path.exe"; Parameters: "/del {app}"

[UninstallDelete]
Type: files; Name: "{app}\hg.exe.local"

[Code]
var
  WriteFile: Boolean;
  CheckDone: Boolean;

function CheckFile(): Boolean;
begin
  if not CheckDone then begin
    WriteFile := True;
    if FileExists(ExpandConstant(CurrentFileName)) then begin
        WriteFile := MsgBox('' + ExpandConstant(CurrentFileName) + '' #13#13 'The file already exists.' #13#13 'Would you like Setup to overwrite it?', mbConfirmation, MB_YESNO) = idYes;
    end;
    CheckDone := True;
  end;
  Result := WriteFile;
end;

procedure ConcatenateFiles();
var
  MergeConfigs: TArrayOfString;
begin
  if LoadStringsFromFile(ExpandConstant('{tmp}\mergetools.hgrc'),MergeConfigs) then begin
    SaveStringsToFile(ExpandConstant(CurrentFileName),MergeConfigs,True);
  end;
end;

procedure Touch(fn: String);
begin
  SaveStringToFile(ExpandConstant(fn), '', False);
end;
