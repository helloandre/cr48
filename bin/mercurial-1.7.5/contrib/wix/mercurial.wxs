<?xml version='1.0' encoding='windows-1252'?>
<Wix xmlns='http://schemas.microsoft.com/wix/2006/wi'>

  <!-- Copyright 2010 Steve Borho <steve@borho.org>

  This software may be used and distributed according to the terms of the
  GNU General Public License version 2 or any later version. -->

  <?include guids.wxi ?>
  <?include defines.wxi ?>

  <?if $(var.Platform) = "x64" ?>
    <?define PFolder = ProgramFiles64Folder ?>
  <?else?>
    <?define PFolder = ProgramFilesFolder ?>
  <?endif?>

  <Product Id='*'
    Name='Mercurial $(var.Version) ($(var.Platform))' 
    UpgradeCode='$(var.ProductUpgradeCode)'
    Language='1033' Codepage='1252' Version='$(var.Version)'
    Manufacturer='Matt Mackall and others'>

    <Package Id='*'
      Keywords='Installer'
      Description="Mercurial distributed SCM (version $(var.Version))"
      Comments='$(var.Comments)'
      Platform='$(var.Platform)'
      Manufacturer='Matt Mackall and others'
      InstallerVersion='300' Languages='1033' Compressed='yes' SummaryCodepage='1252' />

    <Media Id='1' Cabinet='mercurial.cab' EmbedCab='yes' DiskPrompt='CD-ROM #1'
           CompressionLevel='high' />
    <Property Id='DiskPrompt' Value="Mercurial $(var.Version) Installation [1]" />

    <Condition Message='Mercurial MSI installers require Windows XP or higher'>
        VersionNT >= 501
    </Condition>

    <Property Id="INSTALLDIR">
      <ComponentSearch Id='SearchForMainExecutableComponent'
                       Guid='$(var.ComponentMainExecutableGUID)' />
    </Property>

    <!--Property Id='ARPCOMMENTS'>any comments</Property-->
    <Property Id='ARPCONTACT'>mercurial@selenic.com</Property>
    <Property Id='ARPHELPLINK'>http://mercurial.selenic.com/wiki/</Property>
    <Property Id='ARPURLINFOABOUT'>http://mercurial.selenic.com/about/</Property>
    <Property Id='ARPURLUPDATEINFO'>http://mercurial.selenic.com/downloads/</Property>
    <Property Id='ARPHELPTELEPHONE'>http://mercurial.selenic.com/wiki/Support</Property>
    <Property Id='ARPPRODUCTICON'>hgIcon.ico</Property>

    <Property Id='INSTALLEDMERCURIALPRODUCTS' Secure='yes'></Property>
    <Property Id='REINSTALLMODE'>amus</Property>

    <!--Auto-accept the license page-->
    <Property Id='LicenseAccepted'>1</Property>

    <Directory Id='TARGETDIR' Name='SourceDir'>
      <Directory Id='$(var.PFolder)' Name='PFiles'>
        <Directory Id='INSTALLDIR' Name='Mercurial'>
          <Component Id='MainExecutable' Guid='$(var.ComponentMainExecutableGUID)' Win64='$(var.IsX64)'>
            <File Id='hgEXE' Name='hg.exe' Source='dist\hg.exe' KeyPath='yes' />
            <Environment Id="Environment" Name="PATH" Part="last" System="yes"
                         Permanent="no" Value="[INSTALLDIR]bin" Action="set" />
          </Component>
          <Component Id='ReadMe' Guid='$(var.ReadMe.guid)' Win64='$(var.IsX64)'>
              <File Id='ReadMe' Name='ReadMe.html' Source='contrib\win32\ReadMe.html'
                    KeyPath='yes'/>
          </Component>
          <Component Id='COPYING' Guid='$(var.COPYING.guid)' Win64='$(var.IsX64)'>
            <File Id='COPYING' Name='COPYING.rtf' Source='contrib\wix\COPYING.rtf'
                  KeyPath='yes'/>
          </Component>

          <Directory Id='HGRCD' Name='hgrc.d'>
            <Component Id='mercurial.rc' Guid='$(var.mercurial.rc.guid)' Win64='$(var.IsX64)'>
              <File Id='mercurial.rc' Name='Mercurial.rc' Source='contrib\win32\mercurial.ini'
                    ReadOnly='yes' KeyPath='yes'/>
            </Component>
            <Component Id='mergetools.rc' Guid='$(var.mergetools.rc.guid)' Win64='$(var.IsX64)'>
              <File Id='mergetools.rc' Name='MergeTools.rc' Source='contrib\mergetools.hgrc'
                    ReadOnly='yes' KeyPath='yes'/>
            </Component>
            <Component Id='paths.rc' Guid='$(var.paths.rc.guid)' Win64='$(var.IsX64)'>
              <CreateFolder/>
              <IniFile Id="ini0" Action="createLine" Directory="HGRCD" Name="Paths.rc"
                Section="web" Key="cacerts" Value="[INSTALLDIR]hgrc.d\cacert.pem" />
            </Component>
            <Component Id='cacert.pem' Guid='$(var.cacert.pem.guid)' Win64='$(var.IsX64)'>
              <File Id='cacert.pem' Name='cacert.pem' Source='..\misc\cacert.pem'
                    ReadOnly='yes' KeyPath='yes'/>
            </Component>
          </Directory>

          <Directory Id='binFolder' Name='bin'>
            <Component Id='HgCmd' Guid='$(var.hgcmd.guid)' Win64='$(var.IsX64)'>
                <File Id='Hg.Cmd' Name='hg.cmd' KeyPath='yes' Source='contrib\wix\hg.cmd' />
            </Component>
          </Directory>
        </Directory>
      </Directory>

      <Directory Id="ProgramMenuFolder" Name="Programs">
        <Directory Id="ProgramMenuDir" Name="Mercurial $(var.Version)">
          <Component Id="ProgramMenuDir" Guid="$(var.ProgramMenuDir.guid)" Win64='$(var.IsX64)'>
            <RemoveFolder Id='ProgramMenuDir' On='uninstall' />
            <RegistryValue Root='HKCU' Key='Software\Mercurial\InstallDir' Type='string'
                           Value='[INSTALLDIR]' KeyPath='yes' />
            <Shortcut Id='UrlShortcut' Directory='ProgramMenuDir' Name='Mercurial Web Site'
                      Target='[ARPHELPLINK]' Icon="hgIcon.ico" IconIndex='0' />
          </Component>
        </Directory>
      </Directory>

      <?if $(var.Platform) = "x86" ?>
        <Merge Id='VCRuntime' DiskId='1' Language='1033'
              SourceFile='$(var.VCRedistSrcDir)\microsoft.vcxx.crt.x86_msm.msm' />
        <Merge Id='VCRuntimePolicy' DiskId='1' Language='1033'
              SourceFile='$(var.VCRedistSrcDir)\policy.x.xx.microsoft.vcxx.crt.x86_msm.msm' />
      <?else?>
        <Merge Id='VCRuntime' DiskId='1' Language='1033'
              SourceFile='$(var.VCRedistSrcDir)\microsoft.vcxx.crt.x64_msm.msm' />
        <Merge Id='VCRuntimePolicy' DiskId='1' Language='1033'
              SourceFile='$(var.VCRedistSrcDir)\policy.x.xx.microsoft.vcxx.crt.x64_msm.msm' />
      <?endif?>
    </Directory>

    <Feature Id='Complete' Title='Mercurial' Description='The complete package'
        Display='expand' Level='1' ConfigurableDirectory='INSTALLDIR' >
      <Feature Id='MainProgram' Title='Program' Description='Mercurial command line app'
             Level='1' Absent='disallow' >
        <ComponentRef Id='MainExecutable' />
        <ComponentRef Id='distOutput' />
        <ComponentRef Id='ProgramMenuDir' />
        <ComponentRef Id='ReadMe' />
        <ComponentRef Id='COPYING' />
        <ComponentRef Id='mercurial.rc' />
        <ComponentRef Id='mergetools.rc' />
        <ComponentRef Id='paths.rc' />
        <ComponentRef Id='cacert.pem' />
        <ComponentRef Id='helpFolder' />
        <ComponentRef Id='HgCmd' />
        <ComponentGroupRef Id='templatesFolder' />
        <MergeRef Id='VCRuntime' />
        <MergeRef Id='VCRuntimePolicy' />
      </Feature>
      <Feature Id='Locales' Title='Translations' Description='Translations' Level='1'>
        <ComponentGroupRef Id='localeFolder' />
        <ComponentRef Id='i18nFolder' />
      </Feature>
      <Feature Id='Documentation' Title='Documentation' Description='HTML man pages' Level='1'>
        <ComponentGroupRef Id='docFolder' />
      </Feature>
      <Feature Id='Misc' Title='Miscellaneous' Description='Contributed scripts' Level='1'>
        <ComponentGroupRef Id='contribFolder' />
      </Feature>
    </Feature>

    <UIRef Id="WixUI_FeatureTree" />
    <UIRef Id="WixUI_ErrorProgressText" />

    <WixVariable Id="WixUILicenseRtf" Value="contrib\wix\COPYING.rtf" />

    <Icon Id="hgIcon.ico" SourceFile="contrib/win32/mercurial.ico" />

    <Upgrade Id='$(var.ProductUpgradeCode)'>
      <UpgradeVersion
        IncludeMinimum='yes' Minimum='0.0.0' IncludeMaximum='no' OnlyDetect='no'
        Property='INSTALLEDMERCURIALPRODUCTS' />
    </Upgrade>

    <InstallExecuteSequence>
      <RemoveExistingProducts After='InstallInitialize'/>
    </InstallExecuteSequence>

  </Product>
</Wix>
