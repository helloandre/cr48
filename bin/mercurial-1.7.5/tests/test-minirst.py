from pprint import pprint
from mercurial import minirst

def debugformat(title, text, width, **kwargs):
    print "%s formatted to fit within %d characters:" % (title, width)
    print "-" * 70
    formatted = minirst.format(text, width, **kwargs)
    if type(formatted) == tuple:
        print formatted[0]
        print "-" * 70
        pprint(formatted[1])
    else:
        print formatted
    print "-" * 70
    print

paragraphs = """
This is some text in the first paragraph.

  A small indented paragraph.
  It is followed by some lines
  containing random whitespace.
 \n  \n   \nThe third and final paragraph.
"""

debugformat('paragraphs', paragraphs, 60)
debugformat('paragraphs', paragraphs, 30)


definitions = """
A Term
  Definition. The indented
  lines make up the definition.
Another Term
  Another definition. The final line in the
   definition determines the indentation, so
    this will be indented with four spaces.

  A Nested/Indented Term
    Definition.
"""

debugformat('definitions', definitions, 60)
debugformat('definitions', definitions, 30)


literals = r"""
The fully minimized form is the most
convenient form::

  Hello
    literal
      world

In the partially minimized form a paragraph
simply ends with space-double-colon. ::

  ////////////////////////////////////////
  long un-wrapped line in a literal block
  \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\

::

  This literal block is started with '::',
    the so-called expanded form. The paragraph
      with '::' disappears in the final output.
"""

debugformat('literals', literals, 60)
debugformat('literals', literals, 30)


lists = """
- This is the first list item.

  Second paragraph in the first list item.

- List items need not be separated
  by a blank line.
- And will be rendered without
  one in any case.

We can have indented lists:

  - This is an indented list item

  - Another indented list item::

      - A literal block in the middle
            of an indented list.

      (The above is not a list item since we are in the literal block.)

::

  Literal block with no indentation (apart from
  the two spaces added to all literal blocks).

1. This is an enumerated list (first item).
2. Continuing with the second item.

(1) foo
(2) bar

1) Another
2) List

Line blocks are also a form of list:

| This is the first line.
  The line continues here.
| This is the second line.
"""

debugformat('lists', lists, 60)
debugformat('lists', lists, 30)


options = """
There is support for simple option lists,
but only with long options:

--all      Output all.
--both     Output both (this description is
           quite long).
--long     Output all day long.

--par      This option has two paragraphs in its description.
           This is the first.

           This is the second.  Blank lines may be omitted between
           options (as above) or left in (as here).

The next paragraph looks like an option list, but lacks the two-space
marker after the option. It is treated as a normal paragraph:

--foo bar baz
"""

debugformat('options', options, 60)
debugformat('options', options, 30)


fields = """
:a: First item.
:ab: Second item. Indentation and wrapping
     is handled automatically.

Next list:

:small: The larger key below triggers full indentation here.
:much too large: This key is big enough to get its own line.
"""

debugformat('fields', fields, 60)
debugformat('fields', fields, 30)

containers = """
Normal output.

.. container:: debug

   Initial debug output.

.. container:: verbose

   Verbose output.

   .. container:: debug

      Debug output.
"""

debugformat('containers (normal)', containers, 60)
debugformat('containers (verbose)', containers, 60, keep=['verbose'])
debugformat('containers (debug)', containers, 60, keep=['debug'])
debugformat('containers (verbose debug)', containers, 60,
            keep=['verbose', 'debug'])

roles = """Please see :hg:`add`."""
debugformat('roles', roles, 60)


sections = """
Title
=====

Section
-------

Subsection
''''''''''

Markup: ``foo`` and :hg:`help`
------------------------------
"""
debugformat('sections', sections, 20)


admonitions = """
.. note::
   This is a note

   - Bullet 1
   - Bullet 2

   .. warning:: This is a warning Second
      input line of warning

.. danger::
   This is danger
"""

debugformat('admonitions', admonitions, 30)

comments = """
Some text.

.. A comment

   .. An indented comment

   Some indented text.
"""

debugformat('comments', comments, 30)
