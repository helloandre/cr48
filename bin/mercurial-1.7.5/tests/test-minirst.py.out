paragraphs formatted to fit within 60 characters:
----------------------------------------------------------------------
This is some text in the first paragraph.

  A small indented paragraph. It is followed by some lines
  containing random whitespace.

The third and final paragraph.
----------------------------------------------------------------------

paragraphs formatted to fit within 30 characters:
----------------------------------------------------------------------
This is some text in the first
paragraph.

  A small indented paragraph.
  It is followed by some lines
  containing random
  whitespace.

The third and final paragraph.
----------------------------------------------------------------------

definitions formatted to fit within 60 characters:
----------------------------------------------------------------------
A Term
  Definition. The indented lines make up the definition.

Another Term
    Another definition. The final line in the definition
    determines the indentation, so this will be indented
    with four spaces.

  A Nested/Indented Term
    Definition.
----------------------------------------------------------------------

definitions formatted to fit within 30 characters:
----------------------------------------------------------------------
A Term
  Definition. The indented
  lines make up the
  definition.

Another Term
    Another definition. The
    final line in the
    definition determines the
    indentation, so this will
    be indented with four
    spaces.

  A Nested/Indented Term
    Definition.
----------------------------------------------------------------------

literals formatted to fit within 60 characters:
----------------------------------------------------------------------
The fully minimized form is the most convenient form:

  Hello
    literal
      world

In the partially minimized form a paragraph simply ends with
space-double-colon.

  ////////////////////////////////////////
  long un-wrapped line in a literal block
  \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\

  This literal block is started with '::',
    the so-called expanded form. The paragraph
      with '::' disappears in the final output.
----------------------------------------------------------------------

literals formatted to fit within 30 characters:
----------------------------------------------------------------------
The fully minimized form is
the most convenient form:

  Hello
    literal
      world

In the partially minimized
form a paragraph simply ends
with space-double-colon.

  ////////////////////////////////////////
  long un-wrapped line in a literal block
  \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\

  This literal block is started with '::',
    the so-called expanded form. The paragraph
      with '::' disappears in the final output.
----------------------------------------------------------------------

lists formatted to fit within 60 characters:
----------------------------------------------------------------------
- This is the first list item.

  Second paragraph in the first list item.

- List items need not be separated by a blank line.
- And will be rendered without one in any case.

We can have indented lists:

  - This is an indented list item
  - Another indented list item:

      - A literal block in the middle
            of an indented list.

      (The above is not a list item since we are in the literal block.)

  Literal block with no indentation (apart from
  the two spaces added to all literal blocks).

1. This is an enumerated list (first item).
2. Continuing with the second item.
(1) foo
(2) bar
1) Another
2) List

Line blocks are also a form of list:

This is the first line. The line continues here.
This is the second line.
----------------------------------------------------------------------

lists formatted to fit within 30 characters:
----------------------------------------------------------------------
- This is the first list item.

  Second paragraph in the
  first list item.

- List items need not be
  separated by a blank line.
- And will be rendered without
  one in any case.

We can have indented lists:

  - This is an indented list
    item
  - Another indented list
    item:

      - A literal block in the middle
            of an indented list.

      (The above is not a list item since we are in the literal block.)

  Literal block with no indentation (apart from
  the two spaces added to all literal blocks).

1. This is an enumerated list
   (first item).
2. Continuing with the second
   item.
(1) foo
(2) bar
1) Another
2) List

Line blocks are also a form of
list:

This is the first line. The
line continues here.
This is the second line.
----------------------------------------------------------------------

options formatted to fit within 60 characters:
----------------------------------------------------------------------
There is support for simple option lists, but only with long
options:

--all      Output all.
--both     Output both (this description is quite long).
--long     Output all day long.
--par      This option has two paragraphs in its
           description. This is the first.

           This is the second.  Blank lines may be omitted
           between options (as above) or left in (as here).

The next paragraph looks like an option list, but lacks the
two-space marker after the option. It is treated as a normal
paragraph:

--foo bar baz
----------------------------------------------------------------------

options formatted to fit within 30 characters:
----------------------------------------------------------------------
There is support for simple
option lists, but only with
long options:

--all      Output all.
--both     Output both (this
           description is
           quite long).
--long     Output all day
           long.
--par      This option has two
           paragraphs in its
           description. This
           is the first.

           This is the second.
           Blank lines may be
           omitted between
           options (as above)
           or left in (as
           here).

The next paragraph looks like
an option list, but lacks the
two-space marker after the
option. It is treated as a
normal paragraph:

--foo bar baz
----------------------------------------------------------------------

fields formatted to fit within 60 characters:
----------------------------------------------------------------------
a   First item.
ab  Second item. Indentation and wrapping is handled
    automatically.

Next list:

small       The larger key below triggers full indentation
            here.
much too large
            This key is big enough to get its own line.
----------------------------------------------------------------------

fields formatted to fit within 30 characters:
----------------------------------------------------------------------
a   First item.
ab  Second item. Indentation
    and wrapping is handled
    automatically.

Next list:

small       The larger key
            below triggers
            full indentation
            here.
much too large
            This key is big
            enough to get its
            own line.
----------------------------------------------------------------------

containers (normal) formatted to fit within 60 characters:
----------------------------------------------------------------------
Normal output.
----------------------------------------------------------------------

containers (verbose) formatted to fit within 60 characters:
----------------------------------------------------------------------
Normal output.

Verbose output.
----------------------------------------------------------------------
['debug', 'debug']
----------------------------------------------------------------------

containers (debug) formatted to fit within 60 characters:
----------------------------------------------------------------------
Normal output.

Initial debug output.
----------------------------------------------------------------------
['verbose']
----------------------------------------------------------------------

containers (verbose debug) formatted to fit within 60 characters:
----------------------------------------------------------------------
Normal output.

Initial debug output.

Verbose output.

Debug output.
----------------------------------------------------------------------
[]
----------------------------------------------------------------------

roles formatted to fit within 60 characters:
----------------------------------------------------------------------
Please see "hg add".
----------------------------------------------------------------------

sections formatted to fit within 20 characters:
----------------------------------------------------------------------
Title
=====

Section
-------

Subsection
''''''''''

Markup: "foo" and "hg help"
---------------------------
----------------------------------------------------------------------

admonitions formatted to fit within 30 characters:
----------------------------------------------------------------------
Note:
   This is a note

   - Bullet 1
   - Bullet 2

   Warning!
      This is a warning Second
      input line of warning

!Danger!
   This is danger
----------------------------------------------------------------------

comments formatted to fit within 30 characters:
----------------------------------------------------------------------
Some text.

   Some indented text.
----------------------------------------------------------------------

